File: backend/src/main/java/com/equisoft/dca/backend/location/model/PlatformType.java
package com.equisoft.dca.backend.location.model;

import com.equisoft.dca.backend.common.model.BaseEnum;

import lombok.Getter;

@Getter
public enum PlatformType implements BaseEnum<Integer> {

	WINDOWS_UNIX(1, "Windows/UNIX"),
	MAINFRAME(2, "Mainframe"),
	AS400(3, "AS/400"),
	AS400_2(4, "AS/400-2"),
	VSE_BIM(5, "VSE/BIM");

	private final int id;

	private final String name;

	PlatformType(int id, String name) {
		this.id = id;
		this.name = name;
	}

	@Override
	public Integer getId() {
		return id;
	}
}

File: react/src/lib/domain/entities/PlatformType.ts
import BaseEnum from './BaseEnum';

export default class PlatformType extends BaseEnum {
    public static WINDOWS_UNIX = new PlatformType('1', 'location:platformTypeEnum:windowsUnix');
    public static MAINFRAME = new PlatformType('2', 'location:platformTypeEnum:mainframe');
    public static AS400 = new PlatformType('3', 'location:platformTypeEnum:as400');
    public static AS400_2 = new PlatformType('4', 'location:platformTypeEnum:as400-2');
    public static VSE_BIM = new PlatformType('5', 'location:platformTypeEnum:vseBim');

    public constructor(id?: string, name?: string) {
        super(id, name);
    }
}
